spec/wrap.md

```diff
@@ -0,0 +1,104 @@
+# WRAP Specification
+
+WRAP defines the rules for bi-directional communication between a Host and a Wasm module by defining the processes of invocation and subinvocation. It makes use of Host allocated Wasm shared memory to pass encoded buffers between the Host and the Wasm module.
+
+The Wasm module must define the following imports and exports:
+
+#### Exports
+- `_invoke: (bufferLen: u32) => u32`
+    - sends the length of input buffer to Wasm module and it returns the length and the ptr to the new 8 byte long response buffer which contains length of the result buffer + pointer to the result buffer
+
+#### Imports
+- `__subinvoke: (bufferPtr: u32, bufferLen: u32) => u32`
+    - Allows sending any buffer from the Wasm module to the host. Host returns the length of the result buffer to the Wasm module.
```

Comment:
Would be great to specify here that subinvokes are "calls to other wrap module functions, where the invocation method and arguments are packed into a buffer". Being specific about how the buffer is packed here is necessary IMO. Sure this could change, but that'd require a new minor version rev of WRAP (0.3 for ex).

Reply:
Reworded this. It's still not calls to other wrap module functions, it's calls to the host functions.
We have the buffer structure defined below, these are just the export/import signatures and their descriptions.